### PR TITLE
Fixes Objective-C objects memory leaks

### PR DESCRIPTION
**Pull request by zhongwuzw**
Fixes CALayer memory leaks in file FlutterPlatformViews_Internal.mm,FlutterPlatformViews.mm

https://github.com/flutter/flutter/issues/46750

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm [hunk 1]
```diff
@@ -56,9 +56,9 @@ + (CGRect)getCGRectFromSkRect:(const SkRect&)clipSkRect {
 - (void)clipRect:(const SkRect&)clipSkRect {
   CGRect clipRect = [ChildClippingView getCGRectFromSkRect:clipSkRect];
   CGPathRef pathRef = CGPathCreateWithRect(clipRect, nil);
-  CAShapeLayer* clip = [[CAShapeLayer alloc] init];
-  clip.path = pathRef;
-  self.layer.mask = clip;
+  fml::scoped_nsobject<CAShapeLayer> clip([[CAShapeLayer alloc] init]);
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm]: Here and elsewhere, you can just autorelease Objective-C objects.

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm [hunk 2]
```diff
@@ -56,9 +56,9 @@ + (CGRect)getCGRectFromSkRect:(const SkRect&)clipSkRect {
...
+  clip.get().path = pathRef;
+  self.layer.mask = clip.get();
   CGPathRelease(pathRef);
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews_Internal.mm]: Can you convert these pathRefs to use CFRef instead?

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm [hunk 3]
```diff
@@ -493,7 +492,7 @@
   }
   auto contentsScale = flutter_view_.get().layer.contentsScale;
   fml::scoped_nsobject<FlutterOverlayView> overlay_view(
-      [[[FlutterOverlayView alloc] initWithContentsScale:contentsScale] retain]);
+      [[FlutterOverlayView alloc] initWithContentsScale:contentsScale]);
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm]: Good catch!

**Comment by zhongwuzw** [comment]
@chinmaygarde Done.

**Comment by cbracken** [comment]
Waiting for the tree to go green before we merge.

**Comment by joaovirgili** [comment]
is this already available on stable channel?

**Pull request merged**